sep_054.md

```diff
@@ -370,6 +371,39 @@ Package build for distribution:
 * `[root]/.build/ecoli-circuits-standalone-package.ttl`
 
 
+#### Versioning and Release of Packages <a name='versioning'></a>
+
+The time required to build physical instantiations of genetic designs poses a challenge for the versioning and release of packages. 
+On the one hand, one would generally want to check for and correct "cannot be built" bugs before a package is released.
+On the other hand, with current methods building designs has a high cost and often takes weeks or months.
+At the alpha or beta pre-release stages of package development, build issues are to be expected and need not block a release.
+
+For production releases of a package, however, [SemVer](https://semver.org/) "release candidate" versioning SHOULD be used for any package that includes novel materials intended to be built as part of a release process.
+Specifically, a "release candidate" pre-release tag SHOULD be used to indicate the point where an intended release is sent into production.
+
+If all new designs added in a release candidate are able to be built, then the release candidate SHOULD have its status upgraded from release candidate (pre-release) to release (e.g., from `1.0-rc1` to `1.0`).
+If some designs in a release candidate are unable to be built, however, then either the failing materials can be dropped from the package (leaving only known-good materials suitable for release) or the designs can be adjusted to produce a new release candidate.
+
+Note that these policies apply only to designs in a package, not to its dependencies.
+As a matter of practice, however, a production release generally SHOULD NOT have any dependency on any pre-release packages.
+
+##### Example:
+
+Consider an `ecoli-circuits` package at version `1.0`, to which a number of new designs have been added with the intention of producing a `1.1` release.
+When design work is complete, including a check that the designs are expected to be able to be synthesized, the package is tagged for pre-release version `1.1-rc1` and the new designs are sent for synthesis.
+
+Several weeks later, most of the designs have been synthesized successfully, but a few have failed.
+The failing designs are updated to try to resolve their synthesis issues. The package is then tagged as pre-release version `1.1-rc2` and the updated designs are sent for synthesis.
```

Comment:
Not being too familiar with semver, would this be an area why we should consider incrementing a patch number? 1.1-rc1 to 1.1.1-rc1 or is the rc# taking that place?  

I think this highlights an area of confusion for me, as reevaluating a design (not just reordering the previous one) for production, seems like a bugfix. There's also an assumption here that the updated design will perform the same as the previous one, so it can replace it.

On the other hand there also be an assumption that all synthesized products work as intended, were created as intended, or should be distributed/available. So instead, after release, how do we account for bugfixing a package.

In short, in what cases and at what stages are packages fixed and where does versioning reflect that?

Reply:
I think you raise some good points. I would say that the patch number would increment if we're changing for knowledge/software reasons, rc number for build reasons. But then what if you add a logical bugfix to a part e.g. fixing a mutation and add that as a patch. Can that release then only be a release candidate until every part in it has been synthesised? Definitely is easier if we just ignore the physical implementation of parts and just focus on the software, but I do feel like that is really ignoring what makes biology fundamentally different from software.

Reply:
The key distinction here is that an "rc" version is a pre-release, which tells both people and software that its contents are unstable and cannot be counted on to remain the same until the release is finalized. Thus, if we run into problems (logical or physical) with a pre-release, we can fix them without affecting the major/minor version number.  

What this means is that if we find out that part `SuperHairpin` in `1.1-rc1` cannot be synthesized, then we can just drop it and release `1.1` with `SuperHairpin` having been deleted, since `1.1` will still contain all of the materials from `1.0` and thus be backward-compatible. But if we release `1.1` before `SuperHairpin` fails, then the new release that drops it would have to be `2.0` because it is not backward-compatible, having removed materials that were made available in `1.1`.

I've added a note in the discussion section to try to express this reasoning.